p2p/protocol/ping/ping_test.go

```diff
@@ -5,26 +5,29 @@ import (
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/require"
-
 	bhost "github.com/libp2p/go-libp2p/p2p/host/basic"
 	swarmt "github.com/libp2p/go-libp2p/p2p/net/swarm/testing"
 	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
 
 	"github.com/libp2p/go-libp2p-core/peer"
+
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/stretchr/testify/require"
 )
 
 func TestPing(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	h1, err := bhost.NewHost(swarmt.GenSwarm(t), nil)
 	require.NoError(t, err)
+	defer h1.Close()
 	h2, err := bhost.NewHost(swarmt.GenSwarm(t), nil)
 	require.NoError(t, err)
+	defer h2.Close()
 
 	err = h1.Connect(ctx, peer.AddrInfo{
 		ID:    h2.ID(),
-		Addrs: h2.Addrs(),
+		Addrs: []ma.Multiaddr{h2.Addrs()[0]},
```

Comment:
Is there a change we can do that does this at a higher level? I think we see this flakiness in other tests too.

Reply:
True, this is not the first time I'm submitting a patch like this.

At which level would you suggest to do it?